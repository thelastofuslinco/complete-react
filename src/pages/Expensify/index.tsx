import { Component } from 'react'
import { store } from '../../store'
import FilterForm from './components/FilterForm'
import ExpenseForm from './components/ExpenseForm'
import ExpenseList from './components/ExpenseList'
import Header from '../../components/Header'

class Expensify extends Component {
  state = {
    unsubscribe: store.subscribe(() => {
      console.log(store.getState())
    })
  }

  componentWillUnmount(): void {
    this.state.unsubscribe()
  }

  render() {
    return (
      <main className="pageContainer">
        <Header
          title="Expensify app"
          subtitle="Put your expenses in the hands of a computer"
        />
        <div>
          <ExpenseForm />
          <FilterForm />
        </div>
        <ExpenseList />
      </main>
    )
  }
}

export default Expensify
