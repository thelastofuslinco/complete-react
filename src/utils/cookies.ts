export function setCookie(name, value, days) {
  const d = new Date()
  d.setTime(d.getTime() + days * 24 * 60 * 60 * 1000)
  const expires = 'expires=' + d.toUTCString()
  document.cookie = `${name}=${value};${expires};path=/;Secure;HttpOnly;SameSite=Strict`
}

export function getCookie(cookieName) {
  const name = cookieName + '='
  const ca = document.cookie.split(';')

  for (let i = 0; i < ca.length; i++) {
    let c = ca[i]
    while (c.charAt(0) == ' ') {
      c = c.substring(1)
    }
    if (c.indexOf(name) == 0) {
      return c.substring(name.length, c.length)
    }
  }
  return ''
}
