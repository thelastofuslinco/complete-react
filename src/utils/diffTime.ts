export const diffTime = (time) => {
  const date1 = new Date(time).getTime()
  const date2 = new Date().getTime()
  const diff = Math.abs(date2 - date1)
  const diffTime = {
    s: Math.floor(diff / 1000),
    m: Math.floor(diff / (1000 * 60)),
    h: Math.floor(diff / (1000 * 60 * 60))
  }

  if (diffTime.h > 1) {
    return `${diffTime.h} hour ago`
  } else if (diffTime.m > 1) {
    return `${diffTime.m} minutes ago`
  } else {
    return `${diffTime.s} seconds ago`
  }
}
