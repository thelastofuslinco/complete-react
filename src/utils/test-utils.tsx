import { render } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import expensesSlice from '../store/slices/expensesSlice'

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { expenses: expensesSlice },
      preloadedState
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }

  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions })
  }
}
