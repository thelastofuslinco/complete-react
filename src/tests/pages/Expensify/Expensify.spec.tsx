import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithProviders } from '../../../utils/test-utils'
import Expensify from '../../../pages/Expensify'

describe('Expensify page', () => {
  test('should render page', () => {
    renderWithProviders(<Expensify />)
    const element = screen.getByRole('main')
    expect(element).toMatchSnapshot()
  })
})
