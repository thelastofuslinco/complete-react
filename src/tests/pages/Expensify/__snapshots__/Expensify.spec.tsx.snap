// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expensify page should render page 1`] = `
<main
  class="pageContainer"
>
  <form
    class="form"
  >
    <input
      class="input undefined"
      name="description"
      type="text"
      value=""
    />
    <input
      class="input undefined"
      name="amount"
      type="number"
      value=""
    />
    <input
      class="input undefined"
      name="note"
      value=""
    />
    <button
      type="submit"
    >
      send
    </button>
  </form>
  <form
    data=""
  >
    <input
      class="input undefined"
      name="text"
      value=""
    />
    <select
      class="select undefined"
      name="sort"
    >
      <option
        value="amount asc"
      >
        amount asc
      </option>
      <option
        value="amount desc"
      >
        amount desc
      </option>
      <option
        value="description asc"
      >
        description asc
      </option>
      <option
        value="description desc"
      >
        description desc
      </option>
      <option
        value="createdAt asc"
      >
        createdAt asc
      </option>
      <option
        value="createdAt desc"
      >
        createdAt desc
      </option>
    </select>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class=""
          name="date"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      type="submit"
    >
      send
    </button>
  </form>
  <p>
    total:
    0
  </p>
  <ul
    class="expenseList"
  >
    <p>
      No expenses
    </p>
  </ul>
</main>
`;
